File: codigos_exercicios/exercicio18a.m
format long;

while(true)
    
    N=input('\nEscreva o numero de pontos N do somatorio para truncamento ')
    
    if N<0
        warning('esse numero eh negativo, estou saindo')
        break
    end

    % inicializando a soma
    soma = 0;
	
        for n=1:N
            soma = soma + 1/(n^2);
        end

    fprintf('O resultado da soma foi: %f10\nO resultado analitico eh: %f10\n',soma,pi^2/6)
    
end

File: codigos_exercicios/python/exercicio18a.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Exercicio 18a
"""

import numpy as np

while True:

    N = eval(input('\nEscreva o número de pontos N do somatório para truncamento: '))
    
    if N<0:
        print('\nesse número eh negativo, estou saindo')
        break
    
    # inicializando a soma
    soma = 0
    
    for n in range(1,N+1):
        soma += 1/(n**2)
        
    print('O resultado da soma foi ',soma,'\nO resultado analitico eh',(np.pi**2)/6)
        
        
        